lib/gems/pending/util/extensions/miq-module.rb

```diff
@@ -26,15 +26,17 @@ def cache_with_timeout(method, timeout = nil, &block)
         cache = $miq_cache_with_timeout[key]
 
         old_timeout = cache[:timeout]
-        cache.clear if force_reload || (old_timeout && Time.now.utc > old_timeout)
-        break cache[:value] unless cache.empty?
+        timeout_expired = !!old_timeout && (Time.now.utc > old_timeout)
+        break cache[:value] if (!force_reload && !timeout_expired && cache.key?(:value))
```

Comment:
If a request is made to clear the cache we should clear it regardless, so we should keep the force_reload in place.  However, the cache.key?(:value) part should stay.